examples/ConversionForm/BaseInput.tsx

```diff
@@ -0,0 +1,47 @@
+import React from "react";
+
+interface BaseInputProps {
+    dataType: string;
+    handler: (event) => void;
+    options?: [];
+    name: string;
+}
+const BaseInput = (props: BaseInputProps) => {
+    switch (props.dataType) {
```

Comment:
my only concern here is that this switch stuff is going to get worse as you add functionality to the component like validation etc.   I would lean toward just making a component type for each dataType.

Reply:
Still, at some level you will need to map a string datatype to a component. so this is a start